File: frontend/types/mou.ts
export type MOUStatus = "GENERATED" | "EXPORTED";

export type MOU = {
  id: number;
  business_id: number;
  party_a_name: string;
  party_b_name: string;
  purpose: string;
  duration_months: number;
  contribution_details: string;
  revenue_sharing: string;
  dispute_resolution: string;
  cluster_purpose: string;
  draft_text: string;
  pdf_path: string | null;
  status: MOUStatus;
  created_at: string;
  updated_at: string;
};

export type MOUGenerateRequest = {
  business_id: number;
  party_a_name: string;
  party_b_name: string;
  purpose: string;
  duration_months: number;
  contribution_details: string;
  revenue_sharing: string;
  dispute_resolution: string;
  cluster_purpose: string;
};

export type MOUUpdate = Partial<
  Omit<MOUGenerateRequest, "business_id"> & {
    draft_text: string;
  }
>;

File: backend/app/models/mou.py
from datetime import datetime
from typing import Optional

from sqlalchemy import DateTime, ForeignKey, Integer, String, Text, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.db.base import Base


class MOU(Base):
    __tablename__ = "mous"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    business_id: Mapped[int] = mapped_column(ForeignKey("businesses.id"), index=True)
    party_a_name: Mapped[str] = mapped_column(String(255))
    party_b_name: Mapped[str] = mapped_column(String(255))
    purpose: Mapped[str] = mapped_column(Text)
    duration_months: Mapped[int] = mapped_column(Integer)
    contribution_details: Mapped[str] = mapped_column(Text)
    revenue_sharing: Mapped[str] = mapped_column(Text)
    dispute_resolution: Mapped[str] = mapped_column(Text)
    cluster_purpose: Mapped[str] = mapped_column(Text)
    draft_text: Mapped[str] = mapped_column(Text)
    pdf_path: Mapped[Optional[str]] = mapped_column(String(1024), nullable=True)
    status: Mapped[str] = mapped_column(
        String(32),
        default="GENERATED",
        index=True,
    )
    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True),
        server_default=func.now(),
        onupdate=func.now(),
    )

    business: Mapped["Business"] = relationship(back_populates="mous")
